src/script/lua_api/l_noise.cpp

```diff
@@ -595,3 +595,138 @@ const luaL_reg LuaPcgRandom::methods[] = {
 	luamethod(LuaPcgRandom, rand_normal_dist),
 	{0,0}
 };
+
+///////////////////////////////////////
+/*
+	LuaSecureRandom
+*/
+
+void LuaSecureRandom::fill_buff()
+{
+	m_rand_idx = 0;
+
+#ifdef WIN32
+	HCRYPTPROV wctx;
+#else
+	FILE *fp = 0;
+#endif
+
+#ifdef WIN32
```

Comment:
Would be really nice to have this in porting.cpp since it's quite platform independent.